src/openloco/industry.cpp

```diff
@@ -119,69 +119,82 @@ namespace openloco
         if (!(flags & industry_flags::flag_01) && under_construction == 0xFF)
         {
             // Run tile loop for 100 iterations
-            auto obj = object();
             for (int i = 0; i < 100; i++)
             {
-                const auto& surface = tilemgr::get(tile_loop.current()).surface();
-                if (surface != nullptr)
+                sub_45329B(tile_loop.current());
+                if (tile_loop.next() == map_pos())
                 {
-                    if (surface->data()[0] & 0x80)
+                    sub_453354();
+                    break;
+                }
+            }
+        }
+    }
+
+    // 0x0045329B
+    void industry::sub_45329B(const map_pos& pos)
+    {
+        const auto& surface = tilemgr::get(pos).surface();
+        if (surface != nullptr)
+        {
+            if (surface->has_high_type_flag())
+            {
+                if (surface->industry_id() == id())
+                {
+                    uint8_t bl = surface->var_6_SLR5();
+                    auto obj = object();
+                    if (bl == 0 || bl != obj->var_EA)
                     {
-                        if (surface->industry_id() == id())
+                        var_DB++;
+                        if ((!(obj->flags & industry_object_flags::flag_28) && surface->var_4_E0() == 0) || find_5(surface))
```

Comment:
I think this should be `surface->var_4_E0() != 0` (looking at `0x4532F8`)

Reply:
Yep, that seems to fix it. Could you file a PR?